src/main/java/org/mule/service/oauth/internal/AbstractOAuthDancer.java

```diff
@@ -114,16 +134,22 @@ protected AbstractOAuthDancer(String clientId, String clientSecret, String token
   @Override
   public void start() throws MuleException {
     httpClient.start();
+    pollScheduler = schedulerService.ioScheduler(config()
+        .withName(name + "-oauthDancer-tokenRefreshPoll")
+        .withShutdownTimeout(0, MILLISECONDS));
   }
 
   @Override
   public void stop() throws MuleException {
+    if (pollScheduler != null) {
+      pollScheduler.stop();
```

Comment:
use org.mule.runtime.core.internal.util.FunctionalUtils#safely(org.mule.runtime.core.api.util.func.CheckedRunnable) to make sure that the http client is still stopped if this method were to fail for some reason

Reply:
that's internal, not API